src/main/java/com/bina/lrsim/util/ProgramOptions.java

```diff
@@ -0,0 +1,40 @@
+package com.bina.lrsim.util;
+
+import org.apache.log4j.Logger;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+
+/**
+ * Created by bayolau on 6/13/16.
+ */
+abstract public class ProgramOptions {
+
+  private final static Logger log = Logger.getLogger(ProgramOptions.class.getName());
+
+  @Option(name = "--help", aliases = "-h", usage = "print help message")
+  public boolean help;
+
+  public static <Derived extends ProgramOptions> Derived parse(String[] args, Class<Derived> c) {
```

Comment:
@marghoob @gdgib-bina I'm a bit unsure about the optimal JAVA solution. In c++ I can CRTP the child class and thus don't need the ```Class<Derived>``` parameter. But seems like type-erasure in JAVA would require this parameter for instantiation?